### PR TITLE
feat: add private dashboards page to the dashboards section [sc-15489]

### PR DESCRIPTION
**Pull request by ndom91**
## Affected Components
* [ ] Content & Marketing
* [ ] Pricing
* [ ] Test
* [x] Docs
* [ ] Learn
* [ ] Other

## Pre-Requisites
* [ ] Code is linted (`npm run lint`)

## Notes for the Reviewer

- Adds `Dashboards/Private Dashboards` page to detail new authentication procedure and its implications
- Updates `Terraform Provider/Dashboards` page to include the new `isPrivate` boolean config option as well as the `key` value returned to the user

### Todo
- [ ] Clean up screenshot

## Screenshots

image

image

### CODE REVIEW

File: site/content/docs/dashboards/authentication.md [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+title: Private Dashboards
+weight: 48
+menu:
+  resources:
+    parent: "Dashboards"
+---
+
+## Authentication
+
+By default Checkly dashboards are public and visible to anyone who knows the URL. You can now set it to "Private", however, and protect it with an auto-generated password.
```
Comment (drakirnosslin) [inline site/content/docs/dashboards/authentication.md:11]: I'd change the second line to something like: 'You can set a dashboard to private, limiting access to users with a password, or Checkly account,' or similar.
The intent is to make the documentation more 'neutral' from a freshness perspective, 'Now you can' implies to me that the feature is new, which it'll only be for a short while. Let's try to make the wording relevant for as long as we can.

File: site/content/docs/dashboards/authentication.md [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+
+1. Click "Save" to apply the changes.
+2. Next, the "Generate Password" button will appear, click this.
+3. It will open a modal which shows you your new private dashboard password.
+
+Be careful, **this is the only time you will see this password in plaintext**. We do not store the password in plaintext, so if it is ever lost, you will need to regenerate a new one and inform any users of the new credentials.
+
+## Access
+
+After enabling password protection for your dashboard, you can continue to visit it via any custom domain you may have enabled. As well as the default `https://[dashboardId].checkly-dashboards.com` domain. The password authentication will continue to work via both channels.
```
Comment (drakirnosslin) [inline site/content/docs/dashboards/authentication.md:31]: This is missing auth0 access info?

File: site/static/docs/images/dashboards-v2/private_settings-1.png
Comment (drakirnosslin) [inline site/static/docs/images/dashboards-v2/private_settings-1.png:1]: We'll have to update this to not have the beta tag once it's in GA #reminder 😄

File: site/content/docs/dashboards/authentication.md (hunk 2 above)
Comment (ndom91) [reply site/content/docs/dashboards/authentication.md:31]: It is purposely missing / not mentioning Auth0. I thought we were releasing it first with just PW support? Is that not the case? Also I thought it was paid-only as in for `PREPAID` plans as well. Is that not the case? It's enterprise (`CONTRACT`) plans only?

File: site/content/docs/dashboards/authentication.md (hunk 1 above)
Comment (ndom91) [reply site/content/docs/dashboards/authentication.md:11]: Yeah good call, updated :+1:

File: site/static/docs/images/dashboards-v2/private_settings-1.png
Comment (ndom91) [reply site/static/docs/images/dashboards-v2/private_settings-1.png:1]: Yeah its also got the old copy regarding "Password required.." and also renders on the page with a weird black border. I have a "Todo: update screenshot" in the main description above already so I don't forget :wink: :+1:

**Comment by ndom91** [comment]
Copy is :heavy_check_mark: by @drakirnosslin and this PR is good to go whenever we're ready to release private dashboards.

**Review by miliberlin — APPROVED** [review]

**Comment by maxigimenez** [comment]
@ndom91 Great PR! let's hold it until Friday, we should release it together with removing the feature flag

**Pull request merged**